tests/cargo-add.rs

```diff
@@ -943,6 +943,112 @@ fn adds_dependency_with_target_cfg() {
     assert_eq!(val.as_str().unwrap(), "my-package1--CURRENT_VERSION_TEST");
 }
 
+#[test]
+fn adds_features_dependency() {
+    let (_tmpdir, manifest) = clone_out_test("tests/fixtures/add/Cargo.toml.sample");
+
+    // dependency not present beforehand
+    let toml = get_toml(&manifest);
+    assert!(toml["dependencies"].is_none());
+
+    execute_command(
+        &[
+            "add",
+            "https://github.com/killercup/cargo-edit.git",
+            "--features",
+            "jui",
+        ],
+        &manifest,
+    );
+
+    // dependency present afterwards
+    let toml = get_toml(&manifest);
+    let val = toml["dependencies"]["cargo-edit"]["features"][0].as_str();
+    assert_eq!(val, Some("jui"));
+}
+
+#[test]
+fn overrides_existing_features() {
+    overwrite_dependency_test(
+        &["add", "your-face", "--features", "nose"],
+        &["add", "your-face", "--features", "mouth"],
+        r#"
+[dependencies]
+your-face = { version = "your-face--CURRENT_VERSION_TEST", features = ["mouth"] }
+"#,
+    )
+}
+
+#[test]
+fn keeps_existing_features_by_default() {
+    overwrite_dependency_test(
+        &["add", "your-face", "--features", "nose"],
+        &["add", "your-face"],
+        r#"
+[dependencies]
+your-face = { version = "your-face--CURRENT_VERSION_TEST", features = ["nose"] }
+"#,
+    )
+}
+
+#[test]
+fn handles_specifying_features_option_multiple_times() {
+    overwrite_dependency_test(
+        &["add", "your-face"],
+        &[
+            "add",
+            "your-face",
+            "--features",
+            "nose",
+            "--features",
+            "mouth",
+        ],
+        r#"
+[dependencies]
+your-face = { version = "your-face--CURRENT_VERSION_TEST", features = ["nose", "mouth"] }
+"#,
+    )
+}
+
+#[test]
+fn can_be_forced_to_provide_an_empty_features_list() {
+    overwrite_dependency_test(
+        &["add", "your-face"],
+        &["add", "your-face", "--features", ""],
+        r#"
+[dependencies]
+your-face = { version = "your-face--CURRENT_VERSION_TEST", features = [] }
+"#,
+    )
+}
+
+#[test]
+fn parses_space_separated_argument_to_features() {
+    overwrite_dependency_test(
+        &["add", "your-face", "--features", "nose"],
+        &["add", "your-face", "--features", "mouth ears"],
+        r#"
+[dependencies]
+your-face = { version = "your-face--CURRENT_VERSION_TEST", features = ["mouth", "ears"] }
+"#,
+    )
+}
+
+#[test]
+fn only_consumes_one_argument_to_features_option() {
+    // The resulting behaviour is a bit nonsensical.
+    // The alternative would be to force it to explode.
+    overwrite_dependency_test(
+        &["add", "your-face"],
+        &["add", "your-face", "--features", "mouth", "nose"],
+        r#"
+[dependencies]
+your-face = { version = "your-face--CURRENT_VERSION_TEST", features = ["mouth"] }
+nose = { version = "nose--CURRENT_VERSION_TEST", features = ["mouth"] }
+"#,
```

Comment:
hmm, I'm not sure about this, I would expect it add your-face with features "mouth" and "nose" w/o any features, unless specified after that. Alternatively we can return an error if multiple crates are specified if `--features` flag is present.
@killercup wdyt?

Reply:
It looks like we already have a mechanism for forbidding multiple crates when rename or version is specified, so I copied that.